common/stringop.c

```diff
@@ -309,3 +309,10 @@ char *argsep(char **stringp, const char *delim, char *matched) {
 	}
 	return start;
 }
+
+char *strict_regex(const char *str) {
+	size_t size = strlen(str) + 3;
```

Comment:
Using `size_t size = snprintf(NULL, 0, "^%s$", str) + 1` is preferred for new additions